raiden-ts/src/raiden.ts

```diff
@@ -700,35 +727,81 @@ export class Raiden {
     if (!UInt(32).is(amount)) throw new Error('Invalid amount');
 
     const paymentId = !options.paymentId ? undefined : bigNumberify(options.paymentId);
-    if (paymentId && !UInt(8).is(paymentId)) throw new Error('Invalid opts.paymentId');
+    if (paymentId && !UInt(8).is(paymentId)) throw new Error('Invalid options.paymentId');
 
     if (options.secret !== undefined && !Secret.is(options.secret))
-      throw new Error('Invalid opts.secret');
+      throw new Error('Invalid options.secret');
     if (options.secrethash !== undefined && !Hash.is(options.secrethash))
-      throw new Error('Invalid opts.secrethash');
-    let { secret, secrethash } = options;
-    if (!secrethash) {
-      if (!secret) secret = makeSecret();
-      secrethash = getSecrethash(secret);
-    } else if (secret && getSecrethash(secret) !== secrethash) {
-      throw new Error('Secret and secrethash must match if passing both');
-    }
-
-    const promise = this.action$
-      .pipe(
-        filter(isActionOf([transferSigned, transferFailed])),
-        first(action => action.meta.secrethash === secrethash!),
+      throw new Error('Invalid options.secrethash');
+
+    // use provided secret or create one if no secrethash was provided
+    const secret = options.secret
+        ? options.secret
+        : !options.secrethash
+        ? makeSecret()
+        : undefined,
+      secrethash = options.secrethash || getSecrethash(secret!);
+    if (secret && getSecrethash(secret) !== secrethash)
+      throw new Error('Provided secrethash must match the sha256 hash of provided secret');
+
+    const metadata = options.metadata;
+    if (metadata && !Metadata.is(metadata)) throw new Error('Invalid options.metadata');
+
+    return merge(
+      // wait for pathFind response
+      this.action$.pipe(
+        filter(isActionOf([pathFound, pathFindFailed])),
```

Comment:
If we fire the `pathFind` action with the intention of only wanting to see the `estimated_fee` for a path, we would run into trouble right? We will probably need a separate action for this?

Reply:
No. Why would we? The filter below ensures we match only the pathFind/Found/FindFailed, so it's safe to fire them whenever we want and get the response. It just happens that we do always call it on `transfer`, but could be done standalone as well. I hope to create a method to allow querying it oneself as well later.

Reply:
The only point I see is that we don't expose `estimated_fee` yet, but that'll be done when we work on the paid PFS.

Reply:
> I hope to create a method to allow querying it oneself as well later.

That's what I was expecting, just wanted to confirm